roles/pulp/templates/settings.py.j2

```diff
@@ -1,7 +1,21 @@
 CONTENT_ORIGIN="http://{{ ansible_hostname }}:8080"
+API_CONTENT_ORIGIN="http://{{ ansible_hostname }}:24817"
+CONTENT_SERVICE_ORIGIN="http://{{ ansible_hostname }}:24816"
 CACHE_ENABLED=True
 REDIS_HOST="localhost"
 REDIS_PORT=6379
+
+DATABASES = {
+	'default': {
+		'ENGINE': 'django.db.backends.postgresql',
+		'NAME': 'pulp',
+		'USER': 'pulp',
+		'PASSWORD': '{{ pulp_db_password }}',
+		'HOST': 'localhost',
```

Comment:
Right now, the pulp containers are deployed with a dedicated (tunneled/bridged) network, which means "localhost" is the container, not the whole VM we're running in, so it won't be able to reach the database.

For Foreman (and Candlepin), we configure the containers to use "host networking":
https://github.com/theforeman/foreman-quadlet/blob/98f6a4110998d2f36257b9b18415b76571a2f6c4/roles/foreman/tasks/main.yaml#L54
This allows the containers to reach the whole VM and all services running on it. This has the "downside" that you don't need to expose any ports as the container can just open ports on the host.
I am not sure this will trivially work with the Pulp containers (no idea how they expose the various HTTP services they offer), but might make the DB setup easier.